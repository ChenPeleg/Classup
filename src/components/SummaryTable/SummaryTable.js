import React from "react";
import styles from "./SummaryTable.module.scss";
import util from "../../Utility/Utility";

const preCenter = (small, big) => {
  if (small && big) {
  } else return 0;
  let double = Math.floor(100 * (small / big));
  return double;
};
const SummaryTable = (props) => {
  const { q0, q1, q23 } = props.summaryObject.mistakesObject;
  const total = props.summaryObject.numOfquestions;
  let p0 = preCenter(q0, total);
  let p1 = preCenter(q1, total);
  let p23 = preCenter(q23, total);
  const ob = { p0, p1, p23 };
  const pre = util.roundTo100(ob);

  return (
    <>
      <div>
        <table className={styles.sumTable}>
          <thead>
            <tr>
              <th>type of answer</th>
              <th>in numbers</th>
              <th>in precent</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>without errors</td>
              <td>{q0}</td>
              <td>{pre.p0}%</td>
            </tr>
            <tr>
              <td>on second attempt</td>
              <td>{q1}</td>
              <td>{pre.p1}%</td>
            </tr>
            <tr>
              <td>after second attempt</td>
              <td>{q23}</td>
              <td>{pre.p23}%</td>
            </tr>
          </tbody>
        </table>
      </div>
    </>
  );
};
export default SummaryTable;
