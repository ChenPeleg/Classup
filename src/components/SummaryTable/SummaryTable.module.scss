@import "../../assets/styles/base";
.sumTable {
  text-align: center;
  font-size: $Base_fontSize * 0.9;
  border: 1px solid black;
  padding: 3px;
  border-collapse: collapse;
  td,
  th {
    padding: 5px;
    border: inherit;
  }
  td:first-child,
  th:first-child {
    text-align: left;
  }
  tr {
    border: 1px solid black;
  }
}
