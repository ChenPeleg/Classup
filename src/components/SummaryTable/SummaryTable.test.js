
import React from "react";
import { shallow } from "enzyme";
import SummaryTable from "./SummaryTable";
const summaryObjectMock = {
  mistakesObject : { q0: 4, q1: 1, q23: 0 },
  numOfquestions: 5,
}
const wrap = shallow(<SummaryTable summaryObject={summaryObjectMock} />)
describe("<SummaryTable>", () => {
  it("renders correctly", () => { expect(wrap).toMatchSnapshot() })
})