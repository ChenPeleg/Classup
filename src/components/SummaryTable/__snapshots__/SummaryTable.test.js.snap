// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryTable> renders correctly 1`] = `
<Fragment>
  <div>
    <table
      className="sumTable"
    >
      <thead>
        <tr>
          <th>
            type of answer
          </th>
          <th>
            in numbers
          </th>
          <th>
            in precent
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            without errors
          </td>
          <td>
            4
          </td>
          <td>
            80
            %
          </td>
        </tr>
        <tr>
          <td>
            on second attempt
          </td>
          <td>
            1
          </td>
          <td>
            20
            %
          </td>
        </tr>
        <tr>
          <td>
            after second attempt
          </td>
          <td>
            0
          </td>
          <td>
            0
            %
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</Fragment>
`;
